app/scripts/models/auth_brokers/oauth.js

```diff
@@ -1,248 +0,0 @@
-          code: result.code,
-          state: result.state
-        });
-        return result;
-      });
-  },
-
-  /**
-     * Derive scoped keys and encrypt them with the relier's public JWK
-     *
-     * @param {Object} account
-     * @param {String} assertion
-     * @returns {Promise} Returns a promise that resolves into an encrypted bundle
-     * @private
-     */
-  _provisionScopedKeys (account, assertion) {
-    const relier = this.relier;
-    const uid = account.get('uid');
-
-    return Promise.resolve().then(() => {
-      if (account.canFetchKeys()) {
-        // check if requested scopes provide scoped keys
-        return this._oAuthClient.getClientKeyData({
-          assertion: assertion,
-          client_id: relier.get('clientId'), //eslint-disable-line camelcase
-          scope: relier.get('scope')
-        });
-      }
-    }).then((clientKeyData) => {
-      if (! clientKeyData || Object.keys(clientKeyData).length === 0) {
-        // if we got no key data then exit out
-        return null;
-      }
-
-      return account.accountKeys().then((keys) => {
-        return this._scopedKeys.createEncryptedBundle(keys, uid, clientKeyData, relier.get('keysJwk'));
-      });
-    });
-  },
-
-  /**
-     * Overridden by subclasses to provide a strategy to finish the OAuth flow.
-     *
-     * @param {Object} [result] - state sent by OAuth RP
-     * @param {String} [result.state] - state sent by OAuth RP
-     * @param {String} [result.code] - OAuth code generated by the OAuth server
-     * @param {String} [result.redirect] - URL that can be used to redirect to
-     * the RP.
-     *
-     * @returns {Promise}
-     */
-  sendOAuthResultToRelier (/*result*/) {
-    return Promise.reject(new Error('subclasses must override sendOAuthResultToRelier'));
-  },
-
-  finishOAuthSignInFlow (account) {
-    return this.finishOAuthFlow(account, { action: Constants.OAUTH_ACTION_SIGNIN });
-  },
-
-  finishOAuthSignUpFlow (account) {
-    return this.finishOAuthFlow(account, { action: Constants.OAUTH_ACTION_SIGNUP });
-  },
-
-  finishOAuthFlow (account, additionalResultData = {}) {
-    this.session.clear('oauth');
-
-    return this.getOAuthResult(account)
-      .then((result) => {
-        result = _.extend(result, additionalResultData);
-        return this.sendOAuthResultToRelier(result);
-      });
-  },
-
-  persistVerificationData (account) {
-    return Promise.resolve().then(() => {
-      var relier = this.relier;
-      this.session.set('oauth', {
-        access_type: relier.get('access_type'), //eslint-disable-line camelcase
-        action: relier.get('action'),
-        client_id: relier.get('clientId'), //eslint-disable-line camelcase
-        keys: relier.get('keys'),
-        scope: relier.get('scope'),
-        state: relier.get('state')
-      });
-
-      return proto.persistVerificationData.call(this, account);
-    });
-  },
-
-  afterForceAuth (account) {
```

Comment:
All of these `after` methods are needed in the combined broker or else the OAuth flows won't complete correctly, i.e., the user will be stuck on FxA and never make it back to the relier:

<img width="459" alt="screenshot 2019-02-14 at 09 00 45" src="https://user-images.githubusercontent.com/848085/52775218-0d8a1b00-3037-11e9-96ec-f78a8ead10eb.png">